src/components/Wizards/CreateVmWizard/CreateVmWizard.js

```diff
@@ -0,0 +1,315 @@
+      validMsg = this.basicFormFields[target].validate(newValue);
+    }
+    if (this.basicFormFields[target].required && newValue.trim().length === 0) {
+      validMsg = 'is required';
+    }
+
+    if (validMsg) {
+      validMsg = `${this.basicFormFields[target].title} ${validMsg}`;
+    }
+
+    const basicVmSettings = {
+      ...this.state.basicVmSettings,
+      [target]: {
+        value: newValue,
+        validMsg
+      }
+    };
+
+    this.setState(state => ({
+      basicVmSettings: {
+        ...state.basicVmSettings,
+        [target]: {
+          value: newValue,
+          validMsg
+        }
+      }
+    }));
+    this.validateWizard(basicVmSettings);
+  };
+
+  validateWizard = values => {
+    let wizardValid = true;
+
+    // check if all required fields are defined
+    const requiredKeys = Object.keys(this.basicFormFields).filter(key => this.isFieldRequired(key, values));
+    const requiredKeysInValues = Object.keys(values).filter(key => this.isFieldRequired(key, values));
+    if (requiredKeys.length !== requiredKeysInValues.length) {
+      wizardValid = false;
+    }
+
+    // check if all fields are valid
+    for (const key in values) {
+      if (
+        values[key].validMsg &&
+        (this.basicFormFields[key].isVisible ? this.basicFormFields[key].isVisible(values) : true)
+      ) {
+        wizardValid = false;
+        break;
+      }
+    }
+
+    this.setState({
+      wizardValid
+    });
+  };
+
+  getValueFromState = key => this.state[key].map(value => value.metadata.name);
+
+  isFieldRequired = (key, basicVmSettings) => {
+    const field = this.basicFormFields[key];
+    if (field.required) {
+      return field.isVisible ? field.isVisible(basicVmSettings) : true;
+    }
+    return false;
+  };
+
+  getFlavorLabel = () => {
+    if (has(this.state.basicVmSettings, 'flavor.value')) {
+      const flavorValue = this.state.basicVmSettings.flavor.value;
+      if (flavorValue !== CUSTOM_FLAVOR) {
+        return `${TEMPLATE_FLAVOR_LABEL}/${this.state.basicVmSettings.flavor.value}`;
+      }
+    }
+    return undefined;
+  };
+
+  getWorkloadLabel = () => this.getLabel(TEMPLATE_WORKLOAD_LABEL, 'workloadProfile');
+
+  getOsLabel = () => this.getLabel(TEMPLATE_OS_LABEL, 'operatingSystem');
+
+  getLabel = (labelPrefix, value) =>
+    has(this.state.basicVmSettings, value)
+      ? `${labelPrefix}/${get(this.state.basicVmSettings, [value, 'value'])}`
+      : undefined;
+
+  getOperatingSystems = () => {
+    const templates = getTemplatesWithLabels(this.state.templates, [this.getWorkloadLabel(), this.getFlavorLabel()]);
+    return getTemplatesLabelValues(templates, TEMPLATE_OS_LABEL);
+  };
+
+  getWorkloadProfiles = () => {
+    const templates = getTemplatesWithLabels(this.state.templates, [this.getOsLabel(), this.getFlavorLabel()]);
+    return getTemplatesLabelValues(templates, TEMPLATE_WORKLOAD_LABEL);
+  };
+
+  getFlavors = () => {
+    let flavors = [];
```

Comment:
No need to initialize here, can be `const flavors` later.

Reply:
fixed